src/ESM.sol

```diff
@@ -66,67 +52,26 @@ contract ESM is DSNote {
         require(z <= x);
     }
 
-    // -- admin --
-    function file(bytes32 job, address obj) external auth note {
-        if (job == "end") end = EndLike(obj);
-        if (job == "sun") sun = obj;
-    }
-
-    function file(bytes32 job, uint256 val) external auth note {
-        if (job == "cap") cap = val;
-    }
-
-    // -- state changes --
     function fire() external note {
-        require(state == START && full(), "esm/not-fireable");
+        require(!fired,  "esm/already-fired");
+        require(full(),  "esm/min-not-reached");
 
         end.cage();
 
-        state = FIRED;
+        fired = true;
     }
 
-    function free() external auth note {
-        require(state == START || state == FIRED, "esm/not-freeable");
-
-        state = FREED;
-    }
-
-    function burn() external auth note {
-        require(state == START || state == FIRED, "esm/not-burnable");
-
-        sum   = 0;
-        state = BURNT;
-
-        bool ok = gem.transfer(address(sun), gem.balanceOf(address(this)));
-
-        require(ok, "esm/failed-transfer");
-    }
-
-    // -- user actions --
     function join(uint256 wad) external note {
-        require(state == START, "esm/not-joinable");
+        require(!fired, "esm/already-fired");
 
         gems[msg.sender] = add(gems[msg.sender], wad);
         sum = add(sum, wad);
 
-        bool ok = gem.transferFrom(msg.sender, address(this), wad);
-
-        require(ok, "esm/failed-transfer");
-    }
-
-    function exit(address usr, uint256 wad) external note {
-        require(state == FREED, "esm/not-freed");
-
-        gems[msg.sender] = sub(gems[msg.sender], wad);
-        sum = sub(sum, wad);
-
-        bool ok = gem.transfer(usr, wad);
-
-        require(ok, "esm/failed-transfer");
+        require(gem.transferFrom(msg.sender, address(0x0), wad), "esm/transfer-failed");
```

Comment:
I know for the `flap` auctions we are still sending to `address(0)`, but that will eventually change, it is not the pattern we are following for burning the MKR token, as `address(0)` will not affect the `totalSupply`.
I think we have 2 options: leaving the burner address as param or use directly `burn` function (which has to be previously authorized and I don't like much).

Reply:
Yeah agreed, I don't like the idea of `rely`ing the ESM on the MKR token.

Don't know what the current burner contract looks like (IIRC there's one), but I think it should have a publicly accessible `burn` function, so that `burn`ing is not up to governance.